### PR TITLE
Update AOD handling of ambiguous MFT and Fwd tracks 

### PR DESCRIPTION
**Pull request by rpezzi**
* Adds TrackTime and TrackTimeResolution for forward track table
* Ambiguous MFT and forward tracks assigned to BC Slice
* Add comments to Fwd tracks tables element definitions

The PR uses same column definitions for ambiguous as #6970 (and will result in merge conflicts) with it.

### CODE REVIEW

File: Framework/Core/include/Framework/AnalysisDataModel.h [hunk 1]
```diff
@@ -332,17 +332,18 @@ using FullTrack = FullTracks::iterator;
...
-DECLARE_SOA_COLUMN(Z, z, float);                               //!
-DECLARE_SOA_COLUMN(Phi, phi, float);                           //!
-DECLARE_SOA_COLUMN(Tgl, tgl, float);                           //!
-DECLARE_SOA_COLUMN(Signed1Pt, signed1Pt, float);               //!
-DECLARE_SOA_COLUMN(NClusters, nClusters, int8_t);              //!
-DECLARE_SOA_COLUMN(Chi2, chi2, float);                         //!
+DECLARE_SOA_INDEX_COLUMN(Collision, collision); //! Collision to which this track belongs
+DECLARE_SOA_INDEX_COLUMN(BC, bc);               //! BC to which this track belongs
+// TODO change to ForwardTrackTypeEnum when enums are supported
+DECLARE_SOA_COLUMN(TrackType, trackType, uint8_t);             //! Type of track. See enum TrackTypeEnum
```
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]: Shouldn't this be ForwardTrackTypeEnum?

**Comment by jgrosseo** [comment]
Thanks! Can you use also self index (DECLARE_SOA_SELF_INDEX_COLUMN_FULL) for MatchMCHTrackID and a index to the MFT standalone track table for MatchMFTTrackID? Or am I missing something here?

File: Framework/Core/include/Framework/AnalysisDataModel.h (hunk 1 above)
Comment (rpezzi) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: fixed

File: Framework/Core/include/Framework/AnalysisDataModel.h [hunk 2]
```diff
@@ -332,26 +332,29 @@ using FullTrack = FullTracks::iterator;
...
+DECLARE_SOA_COLUMN(Chi2, chi2, float);                                                    //! Track chi^2
+DECLARE_SOA_COLUMN(PDca, pDca, float);                                                    //! PDca for MUONStandalone
+DECLARE_SOA_COLUMN(RAtAbsorberEnd, rAtAbsorberEnd, float);                                //! RAtAbsorberEnd for MUONStandalone tracks and GlobalMuonTrackstracks
+DECLARE_SOA_COLUMN(Chi2MatchMCHMID, chi2MatchMCHMID, float);                              //! MCH-MID Match Chi2 for MUONStandalone tracks
+DECLARE_SOA_COLUMN(Chi2MatchMCHMFT, chi2MatchMCHMFT, float);                              //! MCH-MFT Match Chi2 for GlobalMuonTracks
+DECLARE_SOA_COLUMN(MatchScoreMCHMFT, matchScoreMCHMFT, float);                            //! MCH-MFT Machine Learning Matching Score for GlobalMuonTracks
+DECLARE_SOA_COLUMN(MatchMFTTrackID, matchMFTTrackID, int);                                //! ID of matching MFT track for GlobalMuonTracks
+DECLARE_SOA_SELF_INDEX_COLUMN_FULL(MatchMCHTrack, matchMCHTrack, int, "MatchMCHTrackID"); //! ID of matching MCH track for GlobalMuonTracks
```
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]: In the file this should be the column `fIndexFwdTracks_MatchMCHTrackID`
You can probably reduce MatchMCHTrackID. At least ID can be removed.

File: Framework/Core/include/Framework/AnalysisDataModel.h [hunk 3]
```diff
@@ -332,26 +332,29 @@ using FullTrack = FullTracks::iterator;
...
+DECLARE_SOA_COLUMN(NClusters, nClusters, int8_t);                                         //! Number of clusters
+DECLARE_SOA_COLUMN(Chi2, chi2, float);                                                    //! Track chi^2
+DECLARE_SOA_COLUMN(PDca, pDca, float);                                                    //! PDca for MUONStandalone
+DECLARE_SOA_COLUMN(RAtAbsorberEnd, rAtAbsorberEnd, float);                                //! RAtAbsorberEnd for MUONStandalone tracks and GlobalMuonTrackstracks
+DECLARE_SOA_COLUMN(Chi2MatchMCHMID, chi2MatchMCHMID, float);                              //! MCH-MID Match Chi2 for MUONStandalone tracks
+DECLARE_SOA_COLUMN(Chi2MatchMCHMFT, chi2MatchMCHMFT, float);                              //! MCH-MFT Match Chi2 for GlobalMuonTracks
+DECLARE_SOA_COLUMN(MatchScoreMCHMFT, matchScoreMCHMFT, float);                            //! MCH-MFT Machine Learning Matching Score for GlobalMuonTracks
+DECLARE_SOA_COLUMN(MatchMFTTrackID, matchMFTTrackID, int);                                //! ID of matching MFT track for GlobalMuonTracks
```
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]: Can this be an index column to MFTStandalone?

File: Framework/Core/include/Framework/AnalysisDataModel.h [hunk 4]
```diff
@@ -438,17 +437,22 @@ DECLARE_SOA_EXTENDED_TABLE(MFTTracks, StoredMFTTracks, "MFTTRACK", //!
 
 using MFTTrack = MFTTracks::iterator;
 
+namespace fwdtrack // Index to MFTtrack column must be defined after table definition. Should a SELF_INDEX_COLUMN be used here?
+{
+DECLARE_SOA_INDEX_COLUMN(MFTTrack, matchMFTTrack); //! ID of matching MFT track for GlobalMuonTracks
+}
```
Comment (rpezzi) [inline Framework/Core/include/Framework/AnalysisDataModel.h:444]: @jgrosseo , notice here that in order to define the index to MFT track column, the corresponding mfttrack table must be already defined. This breaks the column definitions in two parts.

**Comment by rpezzi** [comment]
@jgrosseo ,
new commit with change discussed this morning:
* BCId removed from fwdtrack table
* Added ambiguousfwdtracks table with SLICE_INDEX_COLUMN colum for BCs and _ARRAY_INDEX_COLUMN for Collisions.

I understand track-times will be added to central barrel tracks and we would like to have it in the fwd track table as well.

File: Framework/Core/include/Framework/AnalysisDataModel.h (hunk 4 above)
Comment (jgrosseo) [reply Framework/Core/include/Framework/AnalysisDataModel.h:444]: The index is right and also your observation. You can remove "Should a SELF_INDEX_COLUMN be used here?"

**Comment by jgrosseo** [comment]
Thanks for the update. I have merged the PR which keeps only the BC for the ambiguous table, you can merge it with this one, and you need to align only the AMBIGUOUSFWDTR (I already did the mft one). I also have a comment already that in the future we will add the collision indices.

**Comment by jgrosseo** [comment]
The track time update is here https://github.com/AliceO2Group/AliceO2/pull/6986
However, we have to wait for adjusting the Run 3 converter and new data before merging. The same will be for this PR.

**Comment by jgrosseo** [comment]
@rpezzi Can you rebase and align the ambiguous table with the ones in the data model now (BC only)? We also need to adjust the RUN2 converter in AliPhysics

**Comment by rpezzi** [comment]
> @rpezzi Can you rebase and align the ambiguous table with the ones in the data model now (BC only)? We also need to adjust the RUN2 converter in AliPhysics

Hi @jgrosseo , this has been updated. I have included tracktimes in the fwdTrack table, but I see that #6986 has not been merged yet. This info will be readly available after PR #6966 is merged.

**Comment by jgrosseo** [comment]
> > @rpezzi Can you rebase and align the ambiguous table with the ones in the data model now (BC only)? We also need to adjust the RUN2 converter in AliPhysics
> 
> Hi @jgrosseo , this has been updated. I have included tracktimes in the fwdTrack table, but I see that #6986 has not been merged yet. This info will be readly available after PR #6966 is merged.

Thanks. It will be available for RUN3 AODs but not for RUN2 converted files. So the fwdtracks table of RUN2 converted data will be unreadable I think. This has to be adjusted in AliPhysics.

**Comment by rpezzi** [comment]
This PR should be merged in sync with https://github.com/AliceO2Group/O2Physics/pull/84

**Pull request closed without merging**